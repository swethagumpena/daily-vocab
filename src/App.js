import React, { useState } from 'react';
import {
  BrowserRouter, Switch, Route,
} from 'react-router-dom';
import './App.css';
import Index from './components/Index/Index';
import Home from './components/Home/Home';

function App() {
  const [vocab] = useState({ eat: 'I am eating' });
  // {word:sentence}
  return (
    <div className="App">

      <BrowserRouter>
        <Switch>
          <Route path="/" exact>
            <Index />
          </Route>
          <Route path="/home">
            <Home vocab={vocab} />
          </Route>
          <Route path="/add-word">
            {/* <AddWord /> */}
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
