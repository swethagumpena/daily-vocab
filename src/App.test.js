import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import App from './App';

describe(App.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('should render Index component', async () => {
    await waitFor(() => render(<App />));
    expect(screen.getByTestId('Index'));
  });

  // test('should render Home component', async () => {
  //   await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
  //   expect(screen.getByTestId('Home'));
  // });
});
