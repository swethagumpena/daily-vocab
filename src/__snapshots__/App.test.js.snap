// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="page"
      data-testid="Index"
    >
      <div
        class="container"
      >
        <h3>
          daily vocab!
        </h3>
        <p>
          learn a new  word everyday!
        </p>
        <button
          type="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
</div>
`;
